pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/measurements/MeasurementsTNCuda.hpp

```diff
@@ -51,13 +68,69 @@ template <class TensorNetT> class MeasurementsTNCuda {
   private:
     using PrecisionT = typename TensorNetT::PrecisionT;
     using ComplexT = typename TensorNetT::ComplexT;
+    using CFP_t = typename TensorNetT::CFP_t;
 
     const TensorNetT &tensor_network_;
 
   public:
     explicit MeasurementsTNCuda(const TensorNetT &tensor_network)
         : tensor_network_(tensor_network){};
 
+    /**
+     * @brief Probabilities for a subset of the full system.
+     *
+     * @param wires Wires will restrict probabilities to a subset
+     * of the full system.
+     * @param  numHyperSamples Number of hyper samples to use in the calculation
+     * and is default as 1.
+     *
+     * @return Floating point std::vector with probabilities.
+     */
+    template <std::size_t thread_per_block = 256>
+    auto probs(const std::vector<std::size_t> &wires,
+               const int32_t numHyperSamples = 1) -> std::vector<PrecisionT> {
+        PL_ABORT_IF_NOT(std::is_sorted(wires.begin(), wires.end()),
+                        "Invalid wire indices order. Please ensure that the "
+                        "wire indices are in the ascending order.");
+
+        const std::size_t length = std::size_t{1} << wires.size();
+
+        std::vector<PrecisionT> h_res(length, 0.0);
+
+        DataBuffer<CFP_t, int> d_output_tensor(
+            length, tensor_network_.getDevTag(), true);
+
+        d_output_tensor.zeroInit();
+
+        DataBuffer<PrecisionT, int> d_output_probs(
+            length, tensor_network_.getDevTag(), true);
+
+        tensor_network_.get_state_tensor(d_output_tensor.getData(),
+                                         d_output_tensor.getLength(), wires,
+                                         numHyperSamples);
+
+        getProbs_CUDA(d_output_tensor.getData(), d_output_probs.getData(),
+                      length, static_cast<int>(thread_per_block),
+                      tensor_network_.getDevTag().getStreamID());
+
+        PrecisionT sum;
+
+        asum_CUDA_device<PrecisionT>(d_output_probs.getData(), length,
+                                     tensor_network_.getDevTag().getDeviceID(),
+                                     tensor_network_.getDevTag().getStreamID(),
+                                     tensor_network_.getCublasCaller(), &sum);
+
+        PL_ABORT_IF(sum == 0.0, "Sum of probabilities is zero.");
+
+        normalizeProbs_CUDA(d_output_probs.getData(), length, sum,
```

Comment:
We don't need to always offload to the device. What do you think of offloading the execution of probs and normalization to host for "small" args and only patch to the device kernels for considerably large num qubits?

Reply:
Good point. Let's use `256` as a threshold? If the number of elements is less than `256`, let's run calculations on CPUs?